### PR TITLE
Interface view enrichment logic, LF-Value translation changes [DPP-1158]

### PR DESCRIPTION
**Pull request by skisel-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala [hunk 1]
```diff
@@ -378,4 +365,157 @@ final class LfValueTranslation(
...
+  private def toContractArgumentApi(verbose: Boolean)(value: Value): ApiRecord =
+    toApi(verbose, LfEngineToApi.lfValueToApiRecord, "create argument")(value)
+
+  private def toContractKeyApi(verbose: Boolean)(value: Value): ApiValue =
+    toApi(verbose, LfEngineToApi.lfValueToApiValue, "create key")(value)
+
+  private def toInterfaceViewApi(verbose: Boolean, interfaceId: Identifier)(value: Value) =
+    InterfaceView(
+      interfaceId = Some(LfEngineToApi.toApiIdentifier(interfaceId)),
+      viewStatus = Some(ProtoStatus.of(Code.OK.value(), "", Seq.empty)),
```
Comment (skisel-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala:462]: shall we put here some message?

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala [hunk 2]
```diff
@@ -378,4 +365,157 @@ final class LfValueTranslation(
...
+      ec: ExecutionContext,
+      loggingContext: LoggingContext,
+  ): Future[ApiContractData] = {
+
+    val renderResult =
+      eventProjectionProperties.render(witnesses, templateId)
+
+    val verbose = eventProjectionProperties.verbose
+
+    val asyncContractAguments = condFuture(renderResult.contractArguments)(
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala:385]:
```suggestion
    val asyncContractArguments = condFuture(renderResult.contractArguments)(
```
```suggestion
    val asyncContractAguments = condFuture(renderResult.contractArguments)(
```

Comment (skisel-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala:385]: oh wow, thanks

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala [hunk 3]
```diff
@@ -378,4 +365,157 @@ final class LfValueTranslation(
...
+      loggingContext: LoggingContext,
+  ): Future[Value] =
+    condFuture(verbose)(
+      Future(enrich(value)).flatMap(consumeEnricherResult)
+    ).map(_.getOrElse(value))
+
+  private def toApi[T](
+      verbose: Boolean,
+      lfEngineToApiFunction: (Boolean, Value) => Either[String, T],
+      attribute: => String,
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala:446]: Why by name if it's always going to be computed?

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala [hunk 4]
```diff
@@ -264,12 +255,17 @@ final class LfValueTranslation(
...
+    )
+  }
+
+  def engine: Engine = {
+    engineO.getOrElse(
+      sys.error(
+        "LfValueTranslation used to deserialize values in verbose mode without an Engine"
       )
     )
   }
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala:271]: 🙏 
```suggestion
  def engine: Engine = 
    engineO.getOrElse(
      sys.error(
        "LfValueTranslation used to deserialize values in verbose mode without an Engine"
      )
    )
```

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionLogUpdatesConversions.scala [hunk 5]
```diff
@@ -402,49 +402,28 @@ private[events] object TransactionLogUpdatesConversions {
...
+          templateId = Some(LfEngineToApi.toApiIdentifier(createdEvent.templateId)),
+          contractKey = apiContractData.contractKey,
+          createArguments = apiContractData.createArguments,
+          interfaceViews = apiContractData.interfaceViews,
+          witnessParties = requestingParties.view.filter(createdWitnesses(createdEvent)).toSeq,
+          signatories = createdEvent.createSignatories.toSeq,
+          observers = createdEvent.createObservers.toSeq,
+          agreementText = createdEvent.createAgreementText.orElse(Some("")),
+        )
+      )
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionLogUpdatesConversions.scala:426]: nice!

**Review by tudor-da — APPROVED** [review]
Nice 👌

File: ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala (hunk 4 above)
Comment (skisel-da) [reply ledger/participant-integration-api/src/main/scala/platform/store/dao/events/LfValueTranslation.scala:271]: I hate odd braces too, but still sometimes miss it ;) thanks for pointing it out!

**Pull request merged**